### PR TITLE
add stream/tunnel port for cloudcore

### PR DESCRIPTION
**Pull request by allenleu**
add stream/tunnel port for cloudcore

### CODE REVIEW

File: build/cloud/07-deployment.yaml [hunk 1]
```diff
@@ -26,6 +26,12 @@ spec:
             - containerPort: 10000
               name: cloudhub
               protocol: TCP
+            - containerPort: 10003
+              name: stream
+              protocol: TCP
+            - containerPort: 10004
+              name: tunnel
+              protocol: TCP
```
Comment (fisherxu) [inline build/cloud/07-deployment.yaml:34]: I think we can delete the ports here, that's useless here, we have used `hostnetwork` here.

File: build/cloud/ha/03-ha-deployment.yaml.example [hunk 2]
```diff
@@ -52,6 +52,12 @@ spec:
             - containerPort: 10002
               name: certandreadyz
               protocol: TCP
+            - containerPort: 10003
+              name: stream
+              protocol: TCP
+            - containerPort: 10004
+              name: tunnel
+              protocol: TCP
```
Comment (fisherxu) [inline build/cloud/ha/03-ha-deployment.yaml.example:60]: ditto

**Review by allenleu — COMMENTED** [review]
It currently affects the normal startup of cloudcore

File: build/cloud/07-deployment.yaml (hunk 1 above)
Comment (fisherxu) [reply build/cloud/07-deployment.yaml:34]: Can the ports be removed here? cc @zhu733756 Please help take a look :)

Comment (zhu733756) [reply build/cloud/07-deployment.yaml:34]: > Can the ports be removed here? cc @zhu733756 Please help take a look :)

I think it cannot be removed, and the related service type needs to use nodeport or loadblancer.  Therefore, these exposed services can be accessed by the edge nodes.

**Comment by zhu733756** [comment]
@allenleu I have pushed a pr in #3304 to add helm support , we can also discuss there.

File: build/cloud/07-deployment.yaml (hunk 1 above)
Comment (allenleu) [reply build/cloud/07-deployment.yaml:34]: @zhu733756 I just withdrew my submission. Do THE Stream and tunnel ports need to be reserved? Also should readinessGates be removed?

Comment (zhu733756) [reply build/cloud/07-deployment.yaml:34]: > @zhu733756 I just withdrew my submission. Do THE Stream and tunnel ports need to be reserved? Also should readinessGates be removed?

Yeah. And I think you also need to expose nodeport for the cloudcore service.

Comment (allenleu) [reply build/cloud/07-deployment.yaml:34]: @zhu733756 Ok, I need to think about it. There is another problem. I have seen that Cloudcore has functions related to leader-election in the 1.6 version of the code, but the 1.7 version has deleted all the relevant codes. Do you know the reason?

Comment (zhu733756) [reply build/cloud/07-deployment.yaml:34]: > @zhu733756 Ok, I need to think about it. There is another problem. I have seen that Cloudcore has functions related to leader-election in the 1.6 version of the code, but the 1.7 version has deleted all the relevant codes. Do you know the reason?

KubeEdge support ha-cloudcore now . See #2867

Comment (allenleu) [reply build/cloud/07-deployment.yaml:34]: Okey，thanks.

Comment (fisherxu) [reply build/cloud/07-deployment.yaml:34]: @allenleu All the leader selection related code can be removed. :)

Comment (fisherxu) [reply build/cloud/07-deployment.yaml:34]: Also you can take a look for #3304, we can use helm to deploy the cloudcore. @allenleu

**Pull request closed without merging**